lib/bitswap.go

```diff
@@ -45,7 +66,12 @@ func CheckBitswapCID(ctx context.Context, c cid.Cid, ma multiaddr.Multiaddr) (*B
 
 	bs := bsnet.NewFromIpfsHost(h, nilRouter)
 	msg := bsmsg.New(false)
-	msg.AddEntry(c, 0, bsmsgpb.Message_Wantlist_Have, true)
+
+	wantType := bsmsgpb.Message_Wantlist_Have
+	if getBlock {
+		wantType = bsmsgpb.Message_Wantlist_Block
```

Comment:
Maybe it's better to do this in two stages, first check the `have` then check the block. This was pretty quick and easy though. Hopefully transmitting the block should be strictly more helpful than just the `have`